# Лабораторная работа 3.2. Домашнее задание

### Задание №1

Создайте два массива: в первом должны быть четные числа от 2 до 12 включительно, а в другом числа 7, 11, 15, 18, 23, 29.

$1.$ Сложите массивы и возведите элементы получившегося массива в квадрат:

In [2]:
import numpy as np
a = np.arange(2, 13, 2)
b = np.array([7, 11, 15, 18, 23, 29])
print((a + b) ** 2)

[  81  225  441  676 1089 1681]


$2.$ Выведите все элементы из первого массива, индексы которых соответствуют индексам тех элементов второго массива, которые больше 12 и дают остаток 3 при делении на 5.

In [3]:
print(a[(b > 12) & (b % 5 == 3)])

[ 8 10]


*3.* Проверьте условие "Элементы первого массива делятся на 4, элементы второго массива меньше 14". (Подсказка: в результате должен получиться массив с True и False)

In [4]:
print(a % 4 == 0)
print(b < 14)

[False  True False  True False  True]
[ True  True False False False False]


### Задание №2

* Найдите интересный для вас датасет. Например, можно выбрать датасет тут. http://data.un.org/Explorer.aspx (выбираете датасет, жмете на view data, потом download, выбирайте csv формат)
* Рассчитайте подходящие описательные статистики для признаков объектов в выбранном датасете
* Проанализируйте и прокомментируйте содержательно получившиеся результаты
* Все комментарии оформляйте строго в ячейках формата markdown
    

In [24]:
import csv
import numpy as np

with open('musiccaps-public.csv', 'r', newline='') as fi:
    d = csv.reader(fi, delimiter=',')
    st = []
    en = []
    for row in d:
        if row[0] == "ytid":
            continue
        st.append(float(row[1])) #Позиция в видео на YouTube, с которой начинается музыка.
        en.append(float(row[2])) # Позиция в видео YouTube, на которой заканчивается музыка. Все клипы имеют длину 10 секунд.
        
arr1 = np.array(st)
arr2 = np.array(en)

In [25]:
print(f"Средняя позиции в видео на YouTube, с которой начинается музыка.: {round(np.mean(arr1),2)}")
print(f"Средняя позиция в видео YouTube, на которой заканчивается музыка: {round(np.mean(arr2),2)}")

print(f"\nМаксимальное значение позиции в видео на YouTube, с которой начинается музыка.: {np.max(arr1)}")
print(f"Максимальное значение позиции в видео на YouTube, на которой заканчивается музыка: {np.max(arr2)}")

print(f"\nМинимальное значение позиции в видео на YouTube, с которой начинается музыка.: {np.min(arr1)}")
print(f"Минимальное значение позиции в видео на YouTube, на которой заканчивается музыка: {np.min(arr2)}")

print(f"\nСреднее отклонение позиции в видео на YouTube, с которой начинается музыка.: {round(np.std(arr1),2)}")
print(f"Среднее отклонение позиции в видео на YouTube, на которой заканчивается музыка: {round(np.std(arr2),2)}")

Средняя позиции в видео на YouTube, с которой начинается музыка.: 89.1
Средняя позиция в видео YouTube, на которой заканчивается музыка: 99.1

Максимальное значение позиции в видео на YouTube, с которой начинается музыка.: 590.0
Максимальное значение позиции в видео на YouTube, на которой заканчивается музыка: 600.0

Минимальное значение позиции в видео на YouTube, с которой начинается музыка.: 0.0
Минимальное значение позиции в видео на YouTube, на которой заканчивается музыка: 10.0

Среднее отклонение позиции в видео на YouTube, с которой начинается музыка.: 110.22
Среднее отклонение позиции в видео на YouTube, на которой заканчивается музыка: 110.22


1. значение с котрого начинается музыка варьируется от 0 до 590
2. среднее отклонение позиции с которой начинается и на которой заканчивается музыка 110.22